docs/integrations/databases/supabase.mdx

```diff
@@ -3,170 +3,672 @@ title: Integrate Supabase with Clerk
 
 <Callout type="info">
-  Note: Supabase recommends enabling [Row Level Security (RLS)](https://supabase.com/docs/guides/auth/row-level-security) for your database tables and configuring access policies as needed.
+  You must have a local project with Clerk set up already to follow this tutorial. See [the quickstart docs](/docs/quickstarts/overview) to get started.
 </Callout>
 
-After setting those three environment variables, you should be able to start up your application development server.
-
-Install the JavaScript client for Supabase with:
-
-```sh filename="terminal"
-npm install @supabase/supabase-js
-```
-
-You can then initialize the Supabase client by passing it the environment variables and the access token from Clerk.
+<Steps>
+ 
+### Add `user_id` to your Supabase tables
 
-```jsx
-import { useAuth } from '@clerk/clerk-react';
-import { createClient } from "@supabase/supabase-js";
+To show users content scoped to their account, you must create RLS policies that check the user's ID. To access user IDs in your RLS policies, you must add a `user_id` column to the relevant tables:
 
-const supabaseClient = async (supabaseAccessToken) => {
-    const supabase = createClient(process.env.NEXT_PUBLIC_SUPABASE_URL, process.env.NEXT_PUBLIC_SUPABASE_KEY, {
-      global: { headers: { Authorization: `Bearer ${supabaseAccessToken}` } },
-    });
-    // set Supabase JWT on the client object,
-    // so it is sent up with all Supabase requests
-    return supabase;
-  };
-  
-function App() {
-  const { getToken } = useAuth();
+- In your Supabase dashboard, in the sidebar, navigate to **Database** > **Tables**. From here, you can choose which table you want to edit, and add the new column.
+    - You can name the column anything, but for this example, name it `user_id`.
+    - **This column's data type must be `text`**.
 
-  const fetchData = async () => {
-    // TODO #1: Replace with your JWT template name
-    const supabaseAccessToken = await getToken({ template: 'supabase' });
+<Callout type="info">
+  This step is required becase Supabase's `auth.uuid()` function, which normally grants access to the user ID in RLS policies, is not compatible with Clerk's user IDs.
+</Callout>
 
-    const supabase = await supabaseClient(supabaseAccessToken);
+### Create a SQL query that checks the user ID
+
+Next, you must create a `requesting_user_id()` function so you can access the user ID in your RLS policies:
+    1. In your Supabase dashboard, in the sidebar, navigate to **SQL Editor**, then select **New query**. Paste the following into the editor:
+        ```sql
+        create or replace function requesting_user_id() returns text as $$ select nullif(current_setting('request.jwt.claims', true)::json->>'sub', '')::text; $$ language sql stable;
+        ```
+    1. Select **Run** to execute the query and create the function.
```

Comment:
```suggestion
    2. Select **Run** to execute the query and create the function.
```

Reply:
Markdown can auto-number a list, so it's best to use all 1s where possible. Prevents issues if you delete a step in the middle. Here's how it looks in the preview:
![image](https://github.com/clerk/clerk-docs/assets/5156915/4791d573-004f-4739-8323-719b047374fe)